### PR TITLE
Search Enhancements Add API Definition Content Search

### PR DESCRIPTION
**Pull request by Avishka-Shamendra**
## Description

Introduces support for API definition content search
Fixes https://github.com/wso2/api-manager/issues/3103

## Related PRs

https://github.com/wso2/carbon-apimgt/pull/12480
https://github.com/wso2/apim-apps/pull/735

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java [hunk 1]
```diff
@@ -314,19 +308,39 @@ public void testContentSearchWithAccessControl() throws Exception {
             "wso2.am" }, description = "Test content Search with store visibility", dependsOnMethods = "testContentSearchWithAccessControl")
     public void testContentSearchWithStoreVisibility() throws Exception {
 
+        // Set store visibility to role1
+        HttpResponse httpResponse = restAPIPublisher.getAPI(apiId);
+        Gson g = new Gson();
```
Comment (AnuGayan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java:313]:
```suggestion
        APIDTO apiDto = new Gson().fromJson(httpResponse.getData(), APIDTO.class);
```

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java [hunk 2]
```diff
@@ -314,19 +308,39 @@ public void testContentSearchWithAccessControl() throws Exception {
...
+        APIDTO apiDto = g.fromJson(httpResponse.getData(), APIDTO.class);
```
Comment (AnuGayan) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java:314]:
```suggestion
```

Comment (Avishka-Shamendra) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java]: Fixed in [01a7917](https://github.com/wso2/product-apim/pull/13507/commits/01a7917f0e1323d2c8c94b0ffa1f7c71d673a9d5)

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java (hunk 1 above)
Comment (Avishka-Shamendra) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ContentSearchTestCase.java]: Fixed in [01a7917](https://github.com/wso2/product-apim/pull/13507/commits/01a7917f0e1323d2c8c94b0ffa1f7c71d673a9d5)

**Review by AnuGayan — APPROVED** [review]

**Pull request merged**